test/date_time_parser_test.exs

```diff
@@ -154,6 +169,16 @@ defmodule DateTimeParserTest do
       DateTime.from_naive!(~N[2286-11-20T17:46:39.999999], "Etc/UTC")
     )
 
+    test_datetime_parsing(
+      "9999999999.0000000009",
+      DateTime.from_naive!(~N[2286-11-20T17:46:39.000000], "Etc/UTC")
+    )
+
+    test_datetime_parsing(
+      "9999999999.0000009000",
```

Comment:
As a result of [where we `truncate`](https://github.com/taxjar/date_time_parser/pull/16/files#diff-0c651f0cef46a594e7e817decdccd3dcR1https://github.com/taxjar/date_time_parser/pull/16/files#diff-0c651f0cef46a594e7e817decdccd3dcR16) this does not get rounded. It feels a little like it _should_ round, but, naively, rounding results in other problems. Let me know what you think and if we should try to come up with a more clever solution.

Reply:
Rounding feel right, but that's sounds like a different issue for a different PR.